### PR TITLE
feat(open-api-gateway): add python client generation

### PR DESCRIPTION
**Pull request by cogwirrel**
Adds support for `ClientLanguage.PYTHON` for generating python clients.

### CODE REVIEW

File: packages/open-api-gateway/src/project/codegen/components/generated-typescript-client-source-code.ts [hunk 1]
```diff
@@ -47,21 +48,12 @@ export class GeneratedTypescriptClientSourceCode extends Component {
...
-          "..",
-          "..",
-          "..",
-          "scripts",
-          "generators",
-          "typescript"
-        ),
-      }
-    );
+    generateClientCode({
```
Comment (agdimech) [inline packages/open-api-gateway/src/project/codegen/components/generated-typescript-client-source-code.ts:51]: nice abstraction.

File: packages/open-api-gateway/src/project/codegen/components/utils.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+import * as path from "path";
+import { exec } from "projen/lib/util";
+import { ClientLanguage } from "../../languages";
+
+export interface GenerateClientCodeOptions {
```
Comment (agdimech) [inline packages/open-api-gateway/src/project/codegen/components/utils.ts:23]: Missing jsdocs.

File: packages/open-api-gateway/src/project/codegen/generated-python-client.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      ...options,
+    });
+
+    new GeneratedPythonClientSourceCode(this, {
+      specPath: options.specPath,
+    });
+
+    // When in a monorepo, with pip and venv (default), it's useful to install our package into the shared venv to make
+    // it easier for other packages in the monorepo to take dependencies on this package.
+    if (
```
Comment (agdimech) [inline packages/open-api-gateway/src/project/codegen/generated-python-client.ts:55]: Should probably add a comment someone (could even be outside the scope of this package) to mention this approach for using a shared .env to create inter-python deps.

Comment (cogwirrel) [reply packages/open-api-gateway/src/project/codegen/generated-python-client.ts:55]: Good call, I'll add to the README :)

**Review by agdimech — APPROVED** [review]

**Pull request merged**